.github/workflows/verify-repo-tags.yml

```diff
@@ -0,0 +1,80 @@
+name: "Verify Repo Tags"
+
+on:
+  schedule:
+    - cron:  '0 7 * * *'
+  workflow_dispatch: {}
+
+jobs:
+  configure-github:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Verify Repo Tags
+        id: verify_repo_tags
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```

Comment:
Do we know if this token has access to read private repos? The output is different depending on whether or not it does.

Token with private repos enabled:

```
Untagged govuk repos:
govuk-dns-tf

Falsely tagged govuk repos:
content-store-proxy
```

Token without private repos enabled:

```
Untagged govuk repos:
fonts
govuk-account-architecture
govuk-app-deployment-private
govuk-aws-data
govuk-cdn-config-secrets
govuk-dns-tf
govuk-feedback-pipeline
govuk-intent-detector
govuk-paas-office-ip-router
govuk-rota-announcer
govuk-secrets
govuk-user-feedback-processing
govuk-user-reviewer
licensify
whitehall-asset-analysis

Falsely tagged govuk repos:
content-store-proxy
```

We should run with private repo access otherwise the script will falsely report inconsistencies.
May be worth creating a token associated with govuk-ci (credentials are in govuk-secrets) and documenting the token in the README.

Reply:
I've only tested it with my own token, assuming this is the same token used in the user reviewer job.

Reply:
Hmmm looks like GH_TOKEN was created as per these docs: https://github.com/alphagov/govuk-saas-config/tree/main/github#rake-githubconfigure_repos

It's not clear what permissions there are for it 🤔 may be that we need to replace the token post-merge, if the script doesn't work with private repos.